src/main.cpp

```diff
@@ -88,43 +88,33 @@ void printStatus(void) {
 }
 
 void loop() {
-    if (MLX90641_GetFrameData(mlx90641_i2c_addr, frameData) != 0) {
-        Serial.println("Failed to get frame data");
+    const int maxRetries = 5;   
+    int retries = 0;
+    int status;
+
+    do {
+        status = MLX90641_GetFrameData(mlx90641_i2c_addr, frameData);
+        if (status != 0) {
+            retries++;
+            // Serial.printf("Missed frame %d\n", retries);
+            delay(1); // short delay before retry
+        }
+    } while (status != 0 && retries < maxRetries);
```

Comment:
I think it's easier to read: 
```suggestion
    while (!MLX90641_GetFrameData(mlx90641_i2c_addr, frameData) && retries <= maxRetries)
    {
        retries++;
        delay(1);
    }
```